rsa.go

```diff
@@ -377,22 +366,16 @@ func rsaSetCRTParams(key C.GO_RSA_PTR, dmp1, dmq1, iqmp BigInt) bool {
 	return C.go_openssl_RSA_set0_crt_params(key, bigToBN(dmp1), bigToBN(dmq1), bigToBN(iqmp)) == 1
 }
 func newRSAKey3(isPriv bool, n, e, d, p, q, dp, dq, qinv BigInt) (C.GO_EVP_PKEY_PTR, error) {
-	// Construct the parameters.
-	bld := C.go_openssl_OSSL_PARAM_BLD_new()
-	if bld == nil {
-		return nil, newOpenSSLError("OSSL_PARAM_BLD_new")
-	}
-	defer C.go_openssl_OSSL_PARAM_BLD_free(bld)
-
 	type bigIntParam struct {
-		name *C.char
-		num  BigInt
+		name    *C.char
+		num     BigInt
+		private bool
 	}
 
 	comps := make([]bigIntParam, 0, 8)
 
 	required := [...]bigIntParam{
-		{OSSL_PKEY_PARAM_RSA_N, n}, {OSSL_PKEY_PARAM_RSA_E, e}, {OSSL_PKEY_PARAM_RSA_D, d},
+		{_OSSL_PKEY_PARAM_RSA_N, n, false}, {_OSSL_PKEY_PARAM_RSA_E, e, false}, {_OSSL_PKEY_PARAM_RSA_D, d, false},
```

Comment:
It seems like it might be possible to directly call `bld.addBigInt` for each of these rather than temporarily storing them in a slice... maybe worth adding a comment for the slice to point out why this is better if I'm missing something. 😄

Reply:
Good point. There is no need to use the temporary slice. I'll add the big ints directory to the builder :).